### PR TITLE
Add zone edit method

### PR DESCRIPTION
**Pull request by akenn**

### CODE REVIEW

File: lib/zones.js [hunk 1]
```diff
@@ -45,6 +45,21 @@ function read(zid, options) {
...
 
+function edit(zid, options) {
+  if (zid instanceof Zone) {
+    zid = zid.id;
+  }
+
+  var uri = join('zones', zid);
+
+  options.method = 'PATCH';
+  options.body = JSON.stringify(options.body);
```
Comment (akenn) [inline lib/zones.js]: This means using the API will look something like this:
```
client.editZone(zone, {
    body: {
      paused:false
    }
  });
```
I'm not sure if I like that or not. Thoughts?

**Comment by akenn** [comment]
Derp, let me add tests first

File: lib/zones.js (hunk 1 above)
Comment (terinjokes) [inline lib/zones.js]: Look at how [`dns.js`](https://github.com/cloudflare/node-cloudflare/blob/4eabc388845c79dc0a56b98b4ddffaf6685e2597/lib/dns.js#L73) is done. The API expects an edited object to be passed in.

You'll probably need to implement `toJSON` here as well.

File: test/zones.js [hunk 2]
```diff
@@ -101,19 +101,27 @@ test('fetch a Zone by id', async t => {
   t.true(zone instanceof Zone);
 });
 
-test('delete a Zone by id', async t => {
```
Comment (terinjokes) [inline test/zones.js]: You deleted the original test by mistake.

Comment (akenn) [inline test/zones.js]: Oh shoot, sorry! Just re-added

File: README.md [hunk 3]
```diff
@@ -53,6 +53,15 @@ Retrives the list of `Zone` objects, optionally filtered and sorted by `query` o
...
 Retrives the `Zone` for the zone identifier `z_id`.
 
+#### `editZone(z, [options]): Promise<Zone>`
+
+* z: `Zone` object or string zone id
+* options: *see Request Options*
+* *Returns*: A Promise that resolves to a `Zone`
+* *API Reference*: [Zone edit](https://api.cloudflare.com/#zone-edit-zone-properties)
+
+Edit the `paused`, `vanity_name_servers`, and `plan` attributes on a zone.
```
Comment (terinjokes) [inline README.md]: `paused` is documented as being read only.

**Comment by akenn** [comment]
:(

**Pull request closed without merging**